pyflow/host.py

```diff
@@ -176,25 +176,23 @@ def check_cmd(self):
     def run_simple_command(self, cmd):
         raise NotImplementedError
 
-    @property
-    def preamble(self):
+    def preamble(self, exit_hook=[]):
```

Comment:
Using a mutable value as a default for an argument is dangerous. It's a bit more cumbersome, but I'd rather have `None` as a default value, and the set it to `[]` when needed.

```python
def add1(lst=[]):
    lst.append(1)
    return lst
```

```
>>> l = add1()
>>> l
[1]
>>> l2 = add1()
>>> l2
[1, 1]
>>> l
[1, 1]
>>> l is l2
True
```